lib/src/main/java/com/auth0/msg/AbstractMessage.java

```
@@ -0,0 +1,250 @@
+        String jsonMsg = mapper.writeValueAsString(this);
+        if (this.error != null) {
+            throw new InvalidClaimsException("Error present cannot serialize message");
+        }
+        return jsonMsg;
+    }
+
+    /**
+     * @param input the jwt String representation of a message
+     * @param KeyJar that might contain the necessary key
+     */
+    public void fromJwt(String input, KeyJar jar) {
+        this.input = input;
+
+        //This will have logic to parse Jwt to claims
+    }
+
+    /**
+     * Serialize the content of this instance (the claims map) into a jwt string
+     * @param KeyJar the signing keyjar
+     * @param String the algorithm to use in signing the JWT
+     * @return a jwt String
+     * @throws InvalidClaimsException
+     */
+    public String toJwt(KeyJar keyjar, Algorithm algorithm) throws
+            InvalidClaimsException, SerializationException {
+        header.put("alg", algorithm.getName());
+        header.put("typ", "JWT");
+        String signingKeyId = algorithm.getSigningKeyId();
+        if (signingKeyId != null) {
+            header.put("kid", signingKeyId);
+        }
+//        JWTCreator.Builder newBuilder = JWT.create().withHeader(this.header);
+//        for (Claim claimKey: claims.keySet()){
+//            newBuilder.withClaim(claimKey.name, (claimKey.type) claims.get(claimKey));
+//        }
+        return null;
+    }
+
+    /**
+     * Serialize the content of this instance (the claims map) into a jwt string
+     * @param Key the signing key
+     * @param String the algorithm to use in signing the JWT
+     * @return a jwt String
+     * @throws InvalidClaimsException
+     */
+    public String toJwt(Key key, Algorithm algorithm) throws InvalidClaimsException, SerializationException {
+        return null;
+    }
+
+    /**
+     * verify that the required claims are present
+     * @return whether the verification passed
+     */
+    protected boolean verify() {
```

The method signature should tell that this method can throw InvalidClaimsException so that caller can appropriately handle failed validation.

---

Done

---

done